design-proposals/dra.md

```diff
@@ -142,24 +142,31 @@ type DeviceStatus struct {
 type DeviceStatusInfo struct {
 	// Name of the device as specified in spec.domain.devices.gpus.name or spec.domain.devices.hostDevices.name
 	Name string `json:"name"`
-	// DeviceResourceClaimStatus reflects the DRA related information for the device
-	// +optional
-	DeviceResourceClaimStatus *DeviceResourceClaimStatus `json:"deviceResourceClaimStatus,omitempty"`
+  // DeviceResourceClaimStatus reflects the DRA related information for the device
+  DeviceResourceClaimStatus *DeviceResourceClaimStatus `json:"deviceResourceClaimStatus,omitempty"`
 }
 
-// DeviceResourceClaimStatus has to be before SyncVMI call from virt-handler to virt-launcher
 type DeviceResourceClaimStatus struct {
-	// ResourceClaimName is the name of the resource claims object used to provision this resource
+	// ResourceClaimName is the name of the resource claim object used to provision this resource
 	// +optional
-	ResourceClaimName *string `json:"resourceClaimName,omitempty"`
-	// DeviceName is the name of actual device on the host provisioned by the driver as reflected in resourceclaim.status
+	ResourceClaimName string `json:"resourceClaimName"`
+	// DeviceAttributes are the attributes for the allocated device. This information is published by the driver
+	// running on the node in resourceslice.spec.devices.basic.attributes for the allocated device.
 	// +optional
-	DeviceName *string `json:"deviceName,omitempty"`
-	// DeviceAttributes are the attributes published by the driver running on the node in
-	// resourceslice.spec.devices.basic.attributes. The attributes are distinguished by deviceName
-	// and resourceclaim.spec.devices.requests.deviceClassName.
+	DeviceAttributes *DeviceAttributes `json:"deviceAttributes,omitempty"`
```

Comment:
nit: since we are already part of `DeviceResourceClaimStatus`, can we name this field `Attributes`? This will reduce stutter and `Device` prefix is assumed from parent field